functions/index.js

```diff
@@ -45,35 +43,31 @@ const app = actionssdk({
   })
 });
 
-app.intent('actions.intent.MAIN', (conv) => {
+app.intent('actions.intent.MAIN', conv => {
   conv.ask('どうぞ、始めて下さい');
 });
 
-app.intent('actions.intent.TEXT', (conv, input) => {
-  return shiritori.loaded.then(() => {
-    return shiritori.interact(dict, input, conv.data.used)
-        .then(result => {
-          conv.data.used.unshift(input);
-          conv.data.used.unshift(result.word);
-          conv.ask(new SimpleResponse({
-            speech: result.word,
-            text: `${result.word} [${result.kana}]`
-          }));
-        })
-        .catch(result => {
-          if (result.win) {
-            if (result.word) {
-              conv.close(`${result.word} [${result.kana}]`);
-            } else {
-              conv.close('すごい！あなたの勝ちです。');
-            }
-          } else if (result.loose) {
-            conv.close('ざんねん。あなたの負けです。');
-          } else {
-            throw result;
-          }
-        });
-  });
+app.intent('actions.intent.TEXT', async (conv, input) => {
+  const result = await shiritori.interact(dict, input, conv.data.used);
```

Comment:
You don't seem to catch the [`Error`](https://github.com/actions-on-google/actionssdk-shiritori-ja-nodejs/pull/17/files#diff-8febe46e45671febc9bed85e7d64116fR123) anymore.

Reply:
Yep, the previous code was doing nothing but re-throwing it.

Reply:
Shouldn't you at least catch the error here and state an error message? Otherwise the function will crash and so will your action.

Reply:
Done, actually removed the error handling, since it wasn't valid, see https://github.com/actions-on-google/actionssdk-shiritori-ja-nodejs/pull/17/commits/ce5269c789da7a0c6bf1c98e14d86a05bf672b5d commit message for more details.